src/components/charts/AssetSankey.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import { DateTime } from 'luxon';
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import type { ChartDataset } from 'chart.js';
+
+import { Split, Transaction } from '@/book/entities';
+import { useMainCurrency } from '@/hooks/api';
+import { isAsset, isLiability } from '@/book/helpers/accountType';
+import { moneyToString, toFixed } from '@/helpers/number';
+import Sankey from './Sankey';
+
+export type AssetSankeyProps = {
+  guid: string,
+};
+
+export default function AssetSankey({
+  guid,
+}: AssetSankeyProps): JSX.Element {
+  const selectedDate = DateTime.now();
+
+  const { data: byAccount } = useQuery({
+    queryKey: [...Split.CACHE_KEY, guid, 'total-by-account', selectedDate.toISODate()],
+    queryFn: () => Transaction.query(`
+      SELECT
+        splits.account_guid as guid,
+        accounts.name,
+        accounts.account_type as type,
+        SUM(cast(splits.quantity_num as REAL) / splits.quantity_denom) as total
+      FROM transactions AS tx
+      JOIN splits ON splits.tx_guid = tx.guid
+      JOIN accounts ON splits.account_guid = accounts.guid
+      WHERE tx.guid IN (
+        SELECT DISTINCT tx_guid
+        FROM splits
+        WHERE account_guid = '${guid}'
+      )
+      AND tx.post_date >= '${selectedDate.startOf('month').toSQLDate()}'
+      GROUP BY splits.account_guid
+      HAVING SUM(cast(splits.quantity_num as REAL) / splits.quantity_denom) != 0
```

Comment:
This is a useful query for counting total expense for a month in a given account. Probably will extra in the future as some statisticswidget showing Top category spending this month or something like this would be nice in asset account